src/Sentry/TransactionTracer.cs

```diff
@@ -194,12 +194,20 @@ public TransactionTracer(IHub hub, ITransactionContext context)
 
         internal ISpan StartChild(SpanId parentSpanId, string operation)
         {
+            // Limit spans to 1000
+            var isOutOfLimit = _spans.Count >= 1000;
+
             var span = new SpanTracer(_hub, this, parentSpanId, operation)
             {
-                IsSampled = IsSampled
+                IsSampled = !isOutOfLimit
+                    ? IsSampled
+                    : false // sample out out-of-limit spans
             };
 
-            _spans.Add(span);
+            if (!isOutOfLimit)
+            {
+                _spans.Add(span);
+            }
```

Comment:
Can we please log that this happened?
```suggestion
            }
            else
            {
            _hub?.Logger?.LogDebug("Span {0} dropped due to limit reached", span.Description);
```

It'll help users figure out what's going on if "my spans don't show up".

the `_hub?.Logger` is a suggestion, maybe best to just pass the options (since we can swap the logger instance at runtime)